_episodes_rmd/06-best-practices-R.Rmd

```diff
@@ -61,44 +63,44 @@ It is best practice to have the user running the script begin in a consistent di
 
-```{r, eval=FALSE}
+```{r relpath, eval=FALSE}
 dat <- read.csv(file = "files/dataset-2013-01.csv", header = TRUE)
 ```
 
 rather than:
 
-```{r, eval=FALSE}
+```{r abspath, eval=FALSE}
 dat <- read.csv(file = "/Users/Karthik/Documents/sannic-project/files/dataset-2013-01.csv", header = TRUE)
 ```
 
-10. R can run into memory issues. It is a common problem to run out of memory after running R scripts for a long time. To inspect the objects in your current R  environment, you can list the objects, search current packages, and remove objects that are currently not in use. A good practice when running long lines of computationally intensive  code is to remove temporary objects after they have served their purpose. However, sometimes, R will not clean up unused memory for a while after you delete objects. You can force R to tidy up its memory by using `gc()`.
+10. R can run into memory issues. R scripts that run for a long time often run out of memory. To inspect the objects in your current R  environment, you can list the objects, search current packages, and remove objects that are currently not in use. A good practice when running long chunks of computationally intensive  code is to remove temporary objects after they have served their purpose. However, R will not always clean up unused memory immediately after you delete objects. You can force R to tidy up its memory by using `gc()`.
 
-```{r, eval=FALSE}
+```{r gc_ex, eval=FALSE}
 interim_object <- data.frame(rep(1:100,10),rep(101:200,10),rep(201:300,10)) # Sample dataset of 1000 rows
 object.size(interim_object) # Reports the memory size allocated to the object
-rm(interim_object) # Removes only the object itself and not necessarily the memory allotted to it
+rm("interim_object") # Removes only the object itself and not necessarily the memory allotted to it
 gc() # Force R to release memory it is no longer using
 ls()  # Lists all the objects in your current workspace
 rm(list = ls()) # If you want to delete all the objects in the workspace and start with a clean slate
 ```
 
-11. Don't save a session history (the default option in R, when it asks if you want an `RData` file). Instead, start in a clean environment so that older objects don't remain in your environment any longer than they need to. If that happens, it can lead to unexpected results.
+11. Don't save your workspace (the default option in R, when it asks if you want to "Save workspace image [y/n/c]?"). Instead, start in a clean workspace without old objects cluttering it. Leftover objects from previous sessions can lead to unexpected, hard-to-debug results. Do *not* put `rm(list=ls())` (which removes all objects in your current workspace, as shown in the previous code example) at the top of your code, as this is a trap for other users who might `source()` or copy-and-paste your code in the course of their R session. Instead, restart R when you want to start fresh.
```

Comment:
Important to distinguish between *session history* and *workspace*.  (I may accidentally have repeated some of these edits in a separate pull request ...)  Comment about `rm(list=ls())` is an opinion of mine, but I think (??) fairly widely shared.  Maybe worth an issue/discussion.